ormar/models/helpers/sqlalchemy.py

```diff
@@ -317,6 +317,8 @@ def set_constraint_names(meta: "ModelMeta") -> None:
                 f"ix_{meta.tablename}_"
                 f'{"_".join([col for col in constraint._pending_colargs])}'
             )
+        elif isinstance(constraint, sqlalchemy.CheckConstraint) and not constraint.name:
+            constraint.name = f"check_{meta.tablename}_{constraint.sqltext}"
```

Comment:
In some backends names cannot contain special characters and spaces so it cannot be like this, you have to process is to keep only [a-zA-z_]